klippy/extruder_auto_fan.py

```diff
@@ -0,0 +1,39 @@
+# Printer fan support
+#
+# Copyright (C) 2016  Kevin O'Connor <kevin@koconnor.net>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import fan
+
+class ExtruderAutoFan:
+    printer_ready=False
+    
+    def __init__(self,printer,config):
+        self.activation_temp=float(config.get('activation_temp'))
+        
+        self.fan=fan.PrinterFan(printer,config)
+        self.heater = printer.objects.get('extruder0').get_heater()
+        self.toolhead=printer.objects.get('toolhead')
+        
+        printer.reactor.register_timer(self.check_temp,printer.reactor.NOW)
+        
+    def fan_on(self):
+        print_time = self.toolhead.get_last_move_time()
```

Comment:
Unfortunately, it's not valid to call toolhead.get_last_move_time() this way.  The "print time" only makes sense during a print.  To call it this way would result in incorrect flushing of the "look-ahead queue" while in a print and it would effectively disable the "motor off timeout" when not in a print.

What I think would need to happen here would be to add a new interface to the MCU objects to convert a system time (eventtime) to an mcu_time.  Then PrinterFan.set_speed() could be broken into two functions - one that does the actual work and one that just converts from print_time to mcu_time and calls the other function.  This new code could then call the function directly with the calculated mcu_time.

I'll try to put together some code to demo this.